app/forms/idv/finance_form.rb

```diff
@@ -0,0 +1,47 @@
+module Idv
+  class FinanceForm
+    include ActiveModel::Model
+
+    FINANCE_PARAMS = [:ccn, :mortgage, :home_equity_line, :auto_loan].freeze
+
+    attr_reader :idv_params, :user, :finance_type, :finance_account
+
+    validates :finance_type, :finance_account, presence: true
+
+    validate :finance_type_valid
+
+    def initialize(idv_params, user)
+      @idv_params = idv_params
+      @user = user
+      @finance_type = FINANCE_PARAMS.find { |param| idv_params.key? param }
+      @finance_account = idv_params[@finance_type]
+    end
+
+    def submit(params)
+      @finance_type = params[:finance_type]
+      @finance_account = params[:finance_account]
```

Comment:
use `self.finance_type` and `self.finance_account` + a private `attr_writer :finance_type, :finance_account` for consistency with other forms?